tools/src/test/java/org/apache/kafka/tools/consumer/group/DescribeConsumerGroupTest.java

```diff
@@ -16,833 +16,962 @@
+                protocol.name,
+                customConfigs
+        );
+        return ConsumerGroupCommandTestUtils.buildConsumers(
+                1,
+                topicPartitions,
+                () -> new KafkaConsumer<String, String>(configs)
+        );
+    }
 
-        TestUtils.waitForCondition(() -> {
-            Entry<Optional<ConsumerGroupState>, Optional<Collection<PartitionAssignmentState>>> groupOffsets = service.collectGroupOffsets(GROUP);
+    private AutoCloseable consumerGroupClosable(GroupProtocol protocol, String groupId, String topicName, Map<String, Object> customConfigs) {
+        return consumerGroupClosable(protocol, groupId, topicName, customConfigs, 1);
+    }
 
-            Predicate<PartitionAssignmentState> isGrp = s -> Objects.equals(s.group, GROUP);
+    private AutoCloseable consumerGroupClosable(GroupProtocol protocol, String groupId, String topicName, Map<String, Object> customConfigs, int numConsumers) {
+        Map<String, Object> configs = composeConfigs(
+                groupId,
+                protocol.name,
+                customConfigs
+        );
+        return ConsumerGroupCommandTestUtils.buildConsumers(
+                numConsumers,
+                false,
+                topicName,
+                () -> new KafkaConsumer<String, String>(configs)
+        );
+    }
 
-            boolean res = groupOffsets.getKey().map(s -> s.equals(ConsumerGroupState.STABLE)).orElse(false) &&
-                groupOffsets.getValue().isPresent() &&
-                groupOffsets.getValue().get().stream().filter(isGrp).count() == 1;
+    private Map<String, Object> composeConfigs(String groupId, String groupProtocol, Map<String, Object> customConfigs) {
+        Map<String, Object> configs = new HashMap<>();
+        configs.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, clusterInstance.bootstrapServers());
+        configs.put(ConsumerConfig.GROUP_ID_CONFIG, groupId);
+        configs.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
+        configs.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
+        configs.put(ConsumerConfig.GROUP_PROTOCOL_CONFIG, groupProtocol);
+        configs.put(ConsumerConfig.PARTITION_ASSIGNMENT_STRATEGY_CONFIG, RangeAssignor.class.getName());
 
-            if (!res)
-                return false;
+        configs.putAll(customConfigs);
+        return configs;
+    }
 
-            Optional<PartitionAssignmentState> maybeAssignmentState = groupOffsets.getValue().get().stream().filter(isGrp).findFirst();
-            if (!maybeAssignmentState.isPresent())
-                return false;
+    private Runnable describeGroups(ConsumerGroupCommand.ConsumerGroupService service) {
+        return () -> Assertions.assertDoesNotThrow(service::describeGroups);
+    }
+}
 
-            PartitionAssignmentState assignmentState = maybeAssignmentState.get();
+class DescribeConsumerGroupUnitTest {
```

Comment:
multiple classes in single java file is not common. We can move those unit tests to `DescribeConsumerGroupTest`, and than move `ClusterInstance` to method scope